### PR TITLE
Adds SHA512 crypt password hash

### PR DESCRIPTION
**Pull request by awidegreen**
Adds supports for SHA512 password hashes as specified in
https://www.akkadia.org/drepper/SHA-crypt.txt version 0.6

### CODE REVIEW

**Comment by awidegreen** [comment]
A couple of month later ... :)

File: sha-crypt/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+documentation = "https://docs.rs/sha-crypt"
+repository = "https://github.com/RustCrypto/password-hashing"
+keywords = ["crypto", "password", "hashing"]
+edition = "2018"
+categories = ["cryptography"]
+
+[dependencies]
+sha2 = "0.9"
+rand = { version = "0.7",  optional = true }
+constant_time_eq = { version = "0.1", optional = true }
```
Comment (tarcieri) [inline sha-crypt/Cargo.toml:15]: We generally use `subtle`, but given how long this PR has been open I'd be happy to swap it out for you

File: sha-crypt/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+//! on \[1\].
+//!
+//! ```toml
+//! [dependencies]
+//! sha-crypt = { version = "0.1", default-features = false }
+//! ```
+//! # Usage
+//!
+//! ```
+//! extern crate sha_crypt;
```
Comment (tarcieri) [inline sha-crypt/src/lib.rs:11]: Obsolete in 2018 edition

**Review by tarcieri — APPROVED** [review]
@awidegreen sorry, didn't see you updated it.

A couple nits but otherwise looks good, thank you!

**Comment by tarcieri** [comment]
Going to go ahead and merge this.

@awidegreen feel free to address the nits in follow-ups, otherwise I will and will cut a release in the next few days.

**Pull request merged**